apps/coordinator/src/hooks/useTransactionAnalysis.ts

```diff
@@ -0,0 +1,32 @@
+import { useSelector } from "react-redux";
+import { useMemo } from "react";
+import { analyzeTransaction } from "../utils/transactionAnalysisUtils";
+import type { WalletState } from "../selectors/wallet";
+import type { MultisigAddressType } from "@caravan/bitcoin";
+
+export function useTransactionAnalysis() {
+  const {
+    inputs = [],
+    outputs = [],
+    feeRate = 1,
+  } = useSelector((state: any) => state.spend?.transaction || {});
+  const { requiredSigners, totalSigners } = useSelector(
+    (state: WalletState) => state.settings || {},
+  );
+  const addressType = useSelector(
+    (state: WalletState) => state.settings?.addressType,
+  ) as MultisigAddressType;
+
+  return useMemo(
+    () =>
+      analyzeTransaction({
```

Comment:
why not just move all of this analyze transaction code into here? There's a lot of unnecessary redirection I feel like now. 

the dust analysis could be its own hook (not separate functions) and the analyzeTransaction code could just be here and not have to pass around arguments.